src/betterproto/plugin/main.py

```diff
@@ -12,16 +14,19 @@ def main():
     # Read request message from stdin
     data = sys.stdin.buffer.read()
 
-    # Parse request
-    request = plugin.CodeGeneratorRequest()
-    request.ParseFromString(data)
+    request = CodeGeneratorRequest().parse(data)
+
+    # sys.stderr.write(f"request: {request}")
+
+    # TODO: scan request to find Option extensions,
+    #       and if found then monkey patch the relevant messages and parse data again
 
     dump_file = os.getenv("BETTERPROTO_DUMP")
     if dump_file:
         dump_request(dump_file, request)
 
     # Create response
-    response = plugin.CodeGeneratorResponse()
+    response = CodeGeneratorResponse()
```

Comment:
@nat-n I'm inspecting the `response` option here with line 775 of `src/betterproto/lib/google/protobuf/__init__.py` commented out, and I'm seeing:
`request.proto_file` -> 3 files, first one is google types, second is `extension.proto` and 3rd is `double.proto`.
`request.proto_file[2].message_type[0].field[0]` -> this is the `email` field
```
FieldDescriptorProto(name='email', number=3, label=1, type=9, type_name='', extendee='', default_value='', oneof_index=0, json_name='email', options=FieldOptions(ctype=0, packed=False, jstype=0, lazy=False, deprecated=False, weak=False, uninterpreted_option=[]))
```

I unfortunately do not see any information in `FieldOptions` that would suggest that we need to monkey-patch it with the extension reference. My understanding is we would have to have some indication (via the `uninterpreted_option` or something) that we need to monkey patch the extension.

And how would this monkey patching take place? Are we doing:
```python3
request = CodeGeneratorRequest().parse(data)
# somehow figure out that request.proto_file[2].message_type[0].field[0] requires a monkey patch
extension = generate_extenion_dataclass(request.proto_file[1].message_type[0])  # assuming this encapsulates 
request.proto_file[2].message_type[0].field[0].options.load_extension(extension)  # assuming load_extension calls self.__dataclass_fields__...
# and now go through a bytes -> parse loop?
data = bytes(request)
request = CodeGeneratorRequest().parse(data) 
```